base_classes/NXdetector.nxdl.xml

```diff
@@ -686,51 +686,71 @@
     <doc>
-		The value at which the detector goes into saturation. 
-		Especially common to CCD detectors, the data 
-		is known to be invalid above this value.
+    The value at which the detector goes into saturation.
+    Especially common to CCD detectors, the data
+    is known to be invalid above this value.
     </doc>
   </field>
   <field name="number_of_cycles" type="NX_INT">
     <doc>
-      CCD images are sometimes constructed by summing 
-      together multiple short exposures in the 
-      electronics. This reduces background etc. 
-      This is the number of short exposures used to sum 
+      CCD images are sometimes constructed by summing
+      together multiple short exposures in the
+      electronics. This reduces background etc.
+      This is the number of short exposures used to sum
       images for an image.
     </doc>
   </field>
   <field name="sensor_material" type="NX_CHAR">
     <doc>
-       At times, radiation is not directly sensed by the detector.  
-       Rather, the detector might sense the output from some 
-       converter like a scintillator. 
+       At times, radiation is not directly sensed by the detector.
+       Rather, the detector might sense the output from some
+       converter like a scintillator.
        This is the name of this converter material.
     </doc>
   </field>
   <field name="sensor_thickness" type="NX_FLOAT" units="NX_LENGTH">
     <doc>
-       At times, radiation is not directly sensed by the detector.  
-       Rather, the detector might sense the output from some 
-       converter like a scintillator. 
+       At times, radiation is not directly sensed by the detector.
+       Rather, the detector might sense the output from some
+       converter like a scintillator.
        This is the thickness of this converter material.
     </doc>
   </field>
   <field name="threshold_energy" type="NX_FLOAT" units="NX_ENERGY">
     <doc>
-      Single photon counter detectors can be adjusted 
-      for a certain energy range in which they 
-      work optimally. This is the energy setting for this. 
+      Single photon counter detectors can be adjusted
+      for a certain energy range in which they
+      work optimally. This is the energy setting for this.
     </doc>
   </field>
   <group type="NXdetector_module">
     <doc>
-		For use in special cases where the data in NXdetector 
-		is represented in several parts, each with a separate geometry.
+    For use in special cases where the data in NXdetector
+    is represented in several parts, each with a separate geometry.
 
-		Use one or more instances of the NXdetector_module 
-		group to declare regions of interest or some other 
-		subdivision of a detector.
+    Use one or more instances of the NXdetector_module
+    group to declare regions of interest or some other
+    subdivision of a detector.
+    </doc>
+  </group>
+  <group name="pixel_shape" type="NXoff_geometry">
```

Comment:
I believe @zjttoefs is correct.  A definition with a this-or-that component needs support in the XSD (XML Schema for the NXDL).  I'll take a deeper look later today.  There may already be a precedent in another definition.

Reply:
The documentation builds with no errors.

Reply:
Is there a typo on [this line](https://github.com/nexusformat/definitions/compare/master...matthew-d-jones:Add_new_geometry_definitions#diff-ad81bd16e7123a17c8f635aaa45e62acR72)?  Perhaps, instead, this:

    <dim index="2" value="3" />

Reply:
A matter of style in the documentation (in the NXDL files).  

Good practice to keep the first documentation to one line and separate from any succeeding lines.  This allows automatic generation of concise summaries such as for the [base class definitions](http://download.nexusformat.org/doc/html/classes/base_classes/index.html).

Also, please make a line break between 60-80 characters on a line so that the raw source code becomes much easier to read and review [such as this example](https://github.com/nexusformat/definitions/compare/master...matthew-d-jones:Add_new_geometry_definitions#diff-ad81bd16e7123a17c8f635aaa45e62acR72).

Reply:
Considering this [line](https://github.com/nexusformat/definitions/compare/master...matthew-d-jones:Add_new_geometry_definitions#diff-7b003e6b953e03d311dcc7767ac1afb9R87):

    This dataset should only be used only if the ``NXoff_geometry`` group is describing a detector.

Can we describe the algorithm by which the XML Schema could enforce this?  The XML Schema should have no knowledge of any of the NXDL classes.  Probably best not to implement in the XSD since that would introduce this knowledge.

Could we enforce this by some rule in the NXDL file itself?  Where is the information provided by which validation can check this?  By traversing the parent (or chain of parents) of the `NXoff_geometry` instance?

Reply:
That the documentation builds is good, but I'm not sure it is sufficient. The validation may need some help to know that there can be one of two groups under a single name. Even just for the person reading the NXDL `<choice>` make things that little bit more obvious.